src/System.Runtime.Extensions/tests/System/IO/PathTests_Windows.netcoreapp.cs

```diff
@@ -264,5 +264,80 @@ public void GetFullPath_ThrowsOnEmbeddedNulls()
+        public static TheoryData<string, string> TestData_TrimEndingDirectorySeparator => new TheoryData<string, string>
+        {
+            { @"C:\folder\", @"C:\folder" },
+            { @"C:/folder/", @"C:/folder" },
+            { @"/folder/", @"/folder" },
+            { @"\folder\", @"\folder" },
+            { @"folder\", @"folder" },
+            { @"folder/", @"folder" },
+            { @"C:\", @"C:\" },
+            { @"C:/", @"C:/" },
+            { @"", @"" },
+            { @"/", @"/" },
+            { @"\", @"\" },
+            { @"\\server\share\", @"\\server\share" },
+            { @"\\server\share\folder\", @"\\server\share\folder" },
+            { @"\\?\C:\", @"\\?\C:\" },
+            { @"\\?\C:\folder\", @"\\?\C:\folder" },
+            { @"\\?\UNC\", @"\\?\UNC\" },
+            { @"\\?\UNC\a\", @"\\?\UNC\a\" },
+            { @"\\?\UNC\a\folder\", @"\\?\UNC\a\folder" },
+            { null, null }
+        };
+
+        public static TheoryData<string, bool> TestData_EndsInDirectorySeparator => new TheoryData<string, bool>
+        {
+            { @"\", true },
+            { @"/", true },
+            { @"C:\folder\", true },
+            { @"C:/folder/", true },
+            { @"C:\", true },
+            { @"C:/", true },
+            { @"\\", true },
+            { @"//", true },
+            { @"\\server\share\", true },
+            { @"\\?\UNC\a\", true },
+            { @"\\?\C:\", true },
+            { @"\\?\UNC\", true },
+            { @"folder\", true },
+            { @"folder", false },
+            { @"", false },
+            { null, false }
+        };
+
+        [Theory,
+            MemberData(nameof(TestData_TrimEndingDirectorySeparator))]
+        public void TrimEndingDirectorySeparator_String(string path, string expected)
+        {
+            string trimmed = Path.TrimEndingDirectorySeparator(path);
+            Assert.Equal(expected, trimmed);
+            Assert.Same(trimmed, Path.TrimEndingDirectorySeparator(trimmed));
+        }
+
+        [Theory,
+            MemberData(nameof(TestData_TrimEndingDirectorySeparator))]
+        public void TrimEndingDirectorySeparator_ReadOnlySpan(string path, string expected)
+        {
+            ReadOnlySpan<char> trimmed = Path.TrimEndingDirectorySeparator(path.AsSpan());
+            PathAssert.Equal(expected, trimmed);
+            PathAssert.Equal(trimmed, Path.TrimEndingDirectorySeparator(trimmed));
+        }
+
+        [Theory,
+            MemberData(nameof(TestData_EndsInDirectorySeparator))]
+        public void EndsInDirectorySeparator_String(string path, bool expected)
+        {
+            Assert.Equal(expected, Path.EndsInDirectorySeparator(path));
+        }
+
+        [Theory,
+            MemberData(nameof(TestData_EndsInDirectorySeparator))]
+        public void EndsInDirectorySeparator_ReadOnlySpan(string path, bool expected)
+        {
+            Assert.Equal(expected, Path.EndsInDirectorySeparator(path.AsSpan()));
+        }
```

Comment:
It's a little unfortunate that we're exactly duplicating the tests across platforms, just with different test data.  Normally I'd suggest putting the tests into a shared location and then just having the test data be in each of the partial files per platform, but it doesn't look like these test files are structured that way.  It's ok for now then.

Reply:
Yes I attempted to understand how to avoid duplication...but as you said at the moment is not possible without refactor.